src/utils.rs

```diff
@@ -0,0 +1,232 @@
+        debug!("File parsed as JSON: {}", path.display());
+        return Ok(ParsedInput::Json(json_value));
+    }
+
+    // Then try parsing as YAML
+    if let Ok(yaml_value) = serde_yaml::from_str::<serde_yaml::Value>(&content) {
+        debug!("File parsed as YAML: {}", path.display());
+        // Convert YAML to JSON
+        let json_value = serde_json::to_value(&yaml_value)
+            .context("Failed to convert YAML value to JSON value")?;
+        return Ok(ParsedInput::Yaml(json_value));
+    }
+
+    // If neither worked, return an error
+    bail!(
+        "File is neither valid JSON nor valid YAML: {}",
+        path.display()
+    )
+}
+
+/// Gets a JSON string representation of the input file.
+///
+/// Takes ownership of the input path and parses the file only once.
+pub fn get_json_string(input_path: PathBuf) -> Result<String> {
+    let parsed = parse_input_file(&input_path)?;
+
+    match parsed {
+        ParsedInput::Json(json) => {
+            debug!("Using JSON input directly");
+            serde_json::to_string_pretty(&json).context("Failed to convert JSON value to string")
+        }
+        ParsedInput::Yaml(json) => {
+            debug!("Using YAML input converted to JSON");
+            serde_json::to_string_pretty(&json)
+                .context("Failed to convert YAML-derived JSON value to string")
+        }
+    }
+}
+
+/// Gets a path to a JSON file for use with APIs that require a file path.
+///
+/// For JSON inputs, returns the original path directly.
+/// For YAML inputs, converts to JSON and creates a temporary file.
+pub fn get_json_file_path(input_path: PathBuf) -> Result<PathBuf> {
+    match parse_input_file(&input_path)? {
+        ParsedInput::Json(_) => {
+            debug!("Input is already JSON, using original file directly");
+            Ok(input_path)
+        }
+        ParsedInput::Yaml(json) => {
+            debug!("Converting YAML to JSON and creating temporary file");
+            let json_string = serde_json::to_string_pretty(&json)
+                .context("Failed to convert YAML-derived JSON value to string")?;
+
+            let temp_dir = std::env::temp_dir();
+            let temp_file = temp_dir.join("sprocket_temp_input.json");
+
+            fs::write(&temp_file, json_string).context("Failed to write temporary JSON file")?;
+
+            Ok(temp_file)
+        }
+    }
+}
+
+/// Creates a temporary JSON file from the input file contents.
+///
+/// This is used when the underlying API requires a file path rather than
+/// content in memory.
+pub fn create_json_file_from_input(input_path: PathBuf) -> Result<PathBuf> {
+    let json_string = get_json_string(input_path)?;
+
+    let temp_dir = std::env::temp_dir();
+    let temp_file = temp_dir.join("sprocket_temp_input.json");
+
+    fs::write(&temp_file, json_string).context("Failed to write temporary JSON file")?;
+
+    Ok(temp_file)
+}
```

Comment:
This is redundant with `get_json_file_path()`. Just delete this function instead of having both